browser/src/map/Clipboard.js

```diff
@@ -245,51 +248,67 @@
 	// forClipboard: a boolean telling if we need the "Confirm copy to clipboard" link in the end
 	// completeFn: called on completion - with response.
 	// progressFn: allows splitting the progress bar up.
-	_doAsyncDownload: function(type,url,optionalFormData,forClipboard,completeFn,progressFn,onErrorFn) {
-		try {
-			var that = this;
-			var request = new XMLHttpRequest();
+	_doAsyncDownload: async function (
+		type,
+		url,
+		optionalFormData,
+		forClipboard,
+		progressFn,
+	) {
+		if (window.ThisIsAMobileApp)
+			throw new Error(
+				'Attempted to call _doAsyncDownload on a mobile platform, this will never work so should never happen...',
+			);
+
+		var request = new XMLHttpRequest();
```

Comment:
My understanding here is that this probably doesn't matter. It's not executed on a server, and is quite limited (the path should have a /clipboard route appended to it, for instance).

Further, it was here before: this commit refactored the code around here which triggered the warning so I presume someone has analyzed the risk here before me... however still best to get someone else to look over so I'm not dismissing this off-hand. Certainly it's a credible warning: we really do make a request to a server specified on the clipboard, and that is core to the functionality here

Reply:
It will always come from the user.
I think domains can be filtered by optional sanitizer used below: `url = window.processCoolUrl({ url: url, type: 'clipboard' });`
or by blocking some requests using headers defined inside CheckFileInfo.